app/lib/tufts/workflow_status.rb

```diff
@@ -0,0 +1,41 @@
+module Tufts
+  module WorkflowStatus
+    def self.status(id)
+      work_id = id
+      solr_doc_status = solr_status(work_id)
+      if solr_doc_status == 'unpublished' || solr_doc_status == 'deposited'
+        'unpublished'
+      else
+        # Only look for this status in Fedora if necessary
+        published_or_edited(id)
+      end
+    end
+
+    def self.solr_status(id)
+      document = SolrDocument.find(id)
+      begin
+        document['workflow_state_name_ssim'][0]
+      rescue
```

Comment:
can we refactor this? I'm worried about rescuing errors we shouldn't.

Reply:
what would you suggest in this case?

Reply:
It looks to me like we're rescuing from `NoMethodError` on `nil`? Maybe:

```ruby
document = SolrDocument.find(id)
states = document.fetch('workflow_state_name_ssim') { ['unpublished'] }
states.first